### ISSUE TITLE
Create the `Tests` directory elsewhere

### ISSUE THREAD

**Issue by Pierstoval** [sensiolabs/SensioGeneratorBundle#416]
According to symfony/symfony-standard#855 , the `{Bundle}/Tests` directory will be moved to the root `tests/` directory, in the Symfony 3 directory structure.

The problem is that the `generate:bundle` command still creates the `Tests` directory.

What can be done IMO: 

* Create a new `--tests-dir` option.<br>
    ```bash
    $ php bin/console generate:bundle -n --dir=src --namespace=Acme/MyFeatureBundle --bundle-name=AcmeMyFeatureBundle -n
    > Will create src/Acme/MyFeatureBundle/Tests normally
    $ php bin/console generate:bundle -n --dir=src --namespace=Acme/MyFeatureBundle --bundle-name=AcmeMyFeatureBundle --tests-dir=tests
    > Will create tests/AcmeMyFeatureBundle/ and put the default `Controller` directory inside
    ```
    This behavior sounds to me like the `app/Resources/{BundleName}/` structure, so it might be the most understandable to everyone, but of course it constraints the new directory to have the same name as the bundle.

* Add a new `symfony-tests-dir` parameter in the root `composer.json` file, and set it as `tests` by default, and get the information from the command to know where to dump the default tests files.

* Force the tests to be put inside the root `tests` directory and bump another major version of the bundle whenever Symfony3 releases.

What do you think?

**Comment by fabpot** [comment]
I'd prefer the `symfony-tests-dir` approach as this how we dealt with other directory changes.

**Comment by fabpot** [comment]
symfony/symfony-standard#862 makes the needed changes on Symfony SE

**Comment by fabpot** [comment]
Symfony SE exposes the composer variable. Anyone willing to make the PR for this bundle?

**Comment by rbaarsma** [comment]
@fabpot how do you propose to use this `symfony-tests-dir`? Read it directly from `composer.json`? How can we be 100% sure where `composer.json` is? Can we assume that it's in the project's root, 1 or 2 levels under the `kernel.root_dir`? Or could the app dir theoretically be somewhere else entirely..? Basically: *if we should read it directly from composer.json, how would we find it that covers all bases?*

I thought to get the answer to this question from the symfony source, but it seems to me that the `symfony-app-dir`, `symfony-bin-dir`, `symfony-var-dir` and `symfony-web-dir` are not used outside of installation scripts (only in ScriptBuilder.php). Relevant stuff for Symfony is configurable inside `AppKernel.php`, but it doesn't make sense to configure the tests directory there.

I'm not sure about a good solution here. Seems to me that the original poster might've had the best idea. I considered putting it into config.yml, but you'd need a default and if the default isn't correct it wouldn't be backwards compatible (resulting in an error).

The backwards compatability may already be an issue for people generating tests with this bundle as is and then suddenly the directory changes, even though they might not test that new directory. 

So all in all maybe the original poster's idea is the best, although it might easily become a bit 'hidden' feature if it's not documented properly.

**Comment by devsdmf** [comment]
As the symfony default skeleton application, the bundles generated by the Symfony Console also should not use the root tests directory by default ?

**Comment by Pierstoval** [comment]
This is part of the purpose of this issue, but no decision was made about the way to do this

**Comment by devsdmf** [comment]
News ?

**Comment by weaverryan** [comment]
Fixed in #494 without using `symfony-tests-dir` (I've also proposed removing that).

**Comment by devsdmf** [comment]
Awesome!
